src/overlays/actors/ovl_En_Mushi2/z_en_mushi2.c

```diff
@@ -30,134 +45,1192 @@ const ActorInit En_Mushi2_InitVars = {
+        !ENMUSHI2_GET_3(&this->actor)) {
+        func_80A6A024(this);
+    }
+
+    this->unk_358 = 0.0f;
+    this->unk_36A = Rand_S16Offset(240, 40);
+    this->unk_30C = 1;
+
+    if (sp3C != 0) {
+        func_80A69F5C(&this->actor, globalCtx);
+        this->actor.world.rot.y = (u32)Rand_Next() >> 0x10;
+        func_80A6AE14(this);
+    } else {
+        func_80A6A300(this);
+    }
+
+    func_80A68B3C(this);
+}
+
+void EnMushi2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMushi2* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+    func_80A68B6C(this);
+}
+
+void func_80A6A300(EnMushi2* this) {
+    this->unk_368 = 100;
+    this->actor.gravity = -0.25f;
+    this->actor.minVelocityY = -3.5f;
+    this->unk_370 = Rand_S16Offset(-1000, 2000);
+    this->skelAnime.animPlaybackSpeed = 1.5f;
+    this->actionFunc = func_80A6A36C;
+}
+
+void func_80A6A36C(EnMushi2* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp20 = false;
+
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 0.2f);
+    this->actor.velocity.y -= this->actor.velocity.y * D_80A6BA14[ENMUSHI2_GET_3(&this->actor)];
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_80A69424(this, globalCtx);
+    this->actor.shape.rot.y += this->unk_370;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if ((this->actor.floorHeight < this->actor.world.pos.y) &&
+        (this->actor.world.pos.y < (this->actor.floorHeight + 4.0f))) {
+        Math_ScaledStepToS(&this->actor.shape.rot.x, 0, 0x1770);
+        this->actor.shape.rot.z += Rand_S16Offset(-2000, 4000);
+    }
+
+    if (this->actor.floorHeight <= (BGCHECK_Y_MIN + 1)) {
+        sp20 = true;
+        func_80A6A094(this);
+        func_80A6A0D8(this);
+    }
+
+    if (sp20 && (this->unk_36A <= 0)) {
+        Actor_MarkForDeath(&this->actor);
+    } else if (this->actor.bgCheckFlags & 0x20) {
+        func_80A6AAA4(this);
+    } else if (this->actor.bgCheckFlags & 1) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GERUDOFT_WALK);
+        func_80A68F24(this);
+        func_80A691EC(this, this->actor.floorPoly, -1.0f);
+        func_80A69388(this);
+        func_80A6A508(&this->actor);
+    }
+}
+
+void func_80A6A508(Actor* thisx) {
+    EnMushi2* this = THIS;
+
+    if (this->unk_36A > 100) {
+        this->unk_35C = Rand_ZeroOne() + 1.0f;
+    } else {
+        this->unk_35C = (Rand_ZeroOne() * 0.8f) + 1.5f;
+    }
+
+    this->unk_368 = Rand_S16Offset(10, 30);
+    if (this->unk_34C == 0) {
+        func_80A69CE0(thisx);
+    } else {
+        func_80A69ADC(thisx);
+    }
+    this->actionFunc = func_80A6A5C0;
+}
+
+void func_80A6A5C0(EnMushi2* this, GlobalContext* globalCtx) {
+    EnMushi2* this2 = this;
+
+    func_80A69D3C(this);
+    Math_SmoothStepToF(&this->actor.speedXZ, this->unk_35C, 0.1f, 0.5f, 0.0f);
+    if (this->unk_354 < SQ(40.0f)) {
+        f32 temp = 1.0f - ((SQ(40.0f) - this->unk_354) * (1.0f / 16000.0f));
```

Comment:
```suggestion
        f32 temp = 1.0f - ((SQ(40.0f) - this->unk_354) * (1.0f / (10.0f * SQ(40.0f))));
```
This is probably where that number comes from